### PR TITLE
[WIP] Support for go1.8

### LINKED ISSUE #141: go1.8: support for additional features
Many new important features will land in go1.8. Please see document:
https://docs.google.com/document/d/1F778e7ZSNiSmbju3jsEWzShcb8lIO4kDyfKDNm4PNd8/edit?usp=sharing for the full list.

Because these features are new we want drivers to implement them before they are released. Ideally they should be implemented by 2016-12-12 to allow for any driver provided feedback from implementation if necessary.

### PR DESCRIPTION
**Pull request by gernest**
This will eventually close #141`

### CODE REVIEW

**Comment by cznic** [comment]
I think the 'WIP' suggests to do the code review only once the PR is complete, right?

**Comment by gernest** [comment]
@cznic not really, mostly each commit is independent, and you can review as new commits comes in. Only merging has to wait until it is complete.

It will be really helpful to me if we review as I add new features so that by the end I wont have duplicate efforts for the case when I make bad choices.

For Instance, your feedback on the commits which add NamedArg support, can help clear that feature. I intend to work on adding support for canceling queries, which will probably cover the rest of go1.8 features.

**Comment by cznic** [comment]
Okay, thanks for letting me know. The plan is to clear the codereview queue this evening.

File: driver1.8.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
+//+build go1.8
```
Comment (cznic) [inline driver1.8.go]: s/+/ +/

cf.: https://golang.org/pkg/go/build/#hdr-Build_Constraints

**Comment by gernest** [comment]
Thanks. I will now see how I can introduce `context.Context` and enforce canceling of running queries.

**Comment by cznic** [comment]
> I will now see how I can introduce context.Context and enforce canceling of running queries.

The row retrieving closure has to return more == false to stop the query: https://github.com/cznic/ql/blob/eb22df22d0ac223dcd91f16d6de0fc4cfdf768a5/ql.go#L225

**Comment by gernest** [comment]
@cznic  neat trick

[OFFTOPIC] effectively testing the query cancelation I have to slow down queries, in postgres they use `pg_sleep`.

I already laid out the groundwork for supporting something like `pg_sleep` just wondering what will be  a better name for the function `sleep` or `ql_sleep`?

I will submit a separate PR for the function

**Comment by gernest** [comment]
@cznic Can we just merge these changes?

I have hit a wall trying to make queries cancelable. When we come up with a way to halt execution maybe someone can pick it up from there.

**Comment by cznic** [comment]
> Can we just merge these changes?

Yes.

**Pull request merged**